crates/catalog/rest/src/catalog.rs

```diff
@@ -134,6 +134,22 @@ impl RestCatalogConfig {
             Client::builder().default_headers(headers).build()?,
         ))
     }
+
+    fn extract_optional_oauth_params(&self) -> HashMap<&str, &str> {
+        let mut optional_oauth_param = HashMap::new();
+        if let Some(scope) = self.props.get("scope") {
+            optional_oauth_param.insert("scope", scope.as_str());
+        } else {
+            optional_oauth_param.insert("scope", "catalog");
```

Comment:
It doesn't quite make sense to have a default one, since the scope mainly relies on the internal implementation of an OAuth server, but we may still keep it as pyiceberg has it. Open to suggestions.

Reply:
I see java also uses `catalog` as default value, so I think it's fine.

Reply:
Yes, lets keep it, I agree with though @flyrain, we should revisit, the scope part in java and pyiceberg as well. Optional param having a default - also make us introduce special logic for only scope without any apparent benefits.